easybuild/tools/github.py

```diff
@@ -243,10 +243,32 @@ def github_api_get_request(request_f, github_user=None, token=None, **kwargs):
     url = request_f(RestClient(GITHUB_API_URL, username=github_user, token=token))
 
     try:
-        status, data = url.get(**kwargs)
+        status, headers = url.head(**kwargs) 
     except socket.gaierror as err:
         _log.warning("Error occurred while performing get request: %s", err)
-        status, data = 0, None
+        status, headers = 0, None
+
+    links = headers.getheader('Link')
+
+    if links:
+        match = re.search(r"page=(\d+)>; rel=\"last\"", links)
+        last = int(match.groups()[0]) if match else 1
+        data = []
+        for page in range(1, last+1):
+            kwargs['page'] = page
+            try:
+                status, page_data = url.get(**kwargs)
+            except socket.gaierror as err:
+                _log.warning("Error occurred while performing get request: %s", err)
+                status, page_data = 0, None
+                break
+            data.extend(page_data)
+    else:
+        try:
+            status, data = url.get(**kwargs)
+        except socket.gaierror as err:
+            _log.warning("Error occurred while performing get request: %s", err)
+            status, data = 0, None
```

Comment:
@migueldiascosta We can avoid quite a bit of duplicate code here:

```python
if links:
    ...
    pages = range(1, last+1)
else:
    pages = [1]

data = []
for page in pages:
    try:
        status, page_data = url.get(**kwargs)
    except socket.gaierror as err:
        _log.warning("Error occurred while performing get request: %s", err)
        status, page_data = 0, []
        break
    data.extend(page_data)

Reply:
not all API endpoints return lists, we would need something like 

```python
if isinstance(page_data, list):
    data.extend(page_data)
else:
    data = page_data
```
?

Reply:
Ugh, what a mess... :)

This works too:
```python
if not isinstance(page_data, list):
    page_data = [page_data]
data.extend(page_data)

Reply:
@boegel I think this will break the uses of `github_api_get_request` that weren't getting a list to begin with